### PR TITLE
Cleanup some System.Reflection tests

### PR DESCRIPTION
**Pull request by Clockwork-Muse**
Use more idiomatic assertions where possible.

### CODE REVIEW

File: src/System.Reflection/tests/AssemblyNameTests.cs [hunk 1]
```diff
@@ -187,16 +187,20 @@ public void Flags_CurrentlyExecutingAssembly()
...
             AssemblyName assemblyName = new AssemblyName(name);
-            
+
             string extended = $"{expectedName}, Culture=neutral, PublicKeyToken=null";
-            Assert.True(assemblyName.FullName == expectedName || assemblyName.FullName == extended);
+            Assert.StartsWith(expectedName, assemblyName.FullName);
+            if (assemblyName.FullName.Length > expectedName.Length)
+            {
+                Assert.Equal(extended, assemblyName.FullName);
+            }
```
Comment (stephentoub) [inline src/System.Reflection/tests/AssemblyNameTests.cs]: I prefer the old way, as it's much easier for me to understand at a glance what it's checking for.

File: src/System.Reflection/tests/AssemblyNameTests.cs [hunk 2]
```diff
@@ -268,15 +272,19 @@ public void Version(Version version, string versionString)
 
             string expected = "MyAssemblyName, Version=" + versionString;
             string extended = expected + ", Culture=neutral, PublicKeyToken=null";
-            Assert.True(assemblyName.FullName == expected || assemblyName.FullName == extended);
+            Assert.StartsWith(expected, assemblyName.FullName);
+            if (assemblyName.FullName.Length > expected.Length)
+            {
+                Assert.Equal(extended, assemblyName.FullName);
+            }
```
Comment (stephentoub) [inline src/System.Reflection/tests/AssemblyNameTests.cs]: Same comment as on the other case of this.

File: src/System.Reflection/tests/AssemblyTests.cs [hunk 3]
```diff
@@ -68,9 +65,15 @@ public void CustomAttributes(Type type)
...
-            Assert.Equal(expected, result);
+            IEnumerable<Type> customAttrs = Helpers.ExecutingAssembly.DefinedTypes.Select(typeInfo => typeInfo.AsType());
+            if (expected)
+            {
+                Assert.Contains(type, customAttrs);
+            }
+            else
+            {
+                Assert.DoesNotContain(type, customAttrs);
+            }
```
Comment (stephentoub) [inline src/System.Reflection/tests/AssemblyTests.cs]: I prefer the simpler:
```C#
Assert.Equal(expected, customAttrs.Contains(type));
```

File: src/System.Reflection/tests/EventInfoTests.cs [hunk 4]
```diff
@@ -34,8 +34,16 @@ public void IsMulticast_ReturnsTrue(Type type, string name)
...
+            if (isVisible)
+            {
+                Assert.NotNull(eventInfo.GetAddMethod());
+                Assert.NotNull(eventInfo.GetAddMethod(false));
+            }
+            else
+            {
+                Assert.Null(eventInfo.GetAddMethod());
+                Assert.Null(eventInfo.GetAddMethod(false));
+            }
```
Comment (stephentoub) [inline src/System.Reflection/tests/EventInfoTests.cs]: In general I'm not seeing the value of expanding out these asserts like this.  I could see in this case inverting both sides to involve fewer logical negatives, e.g.
```C#
Assert.Equal(isVisible, eventInfo.GetAddMethod() != null);
Assert.Equal(isVisible, eventInfo.GetAddMethod(false) != null);
```
but I don't think we want to double the asserts like this in all such cases.

File: src/System.Reflection/tests/PropertyInfoTests.cs [hunk 5]
```diff
@@ -40,8 +40,22 @@ public void GetConstantValue_NotConstant_ThrowsInvalidOperationException(Type ty
...
+                Assert.Null(propertyInfo.GetMethod);
+            }
+            if (hasSetter)
+            {
+                Assert.NotNull(propertyInfo.SetMethod);
+            }
+            else
+            {
+                Assert.Null(propertyInfo.SetMethod);
+            }
```
Comment (stephentoub) [inline src/System.Reflection/tests/PropertyInfoTests.cs]: Same as the other comments

**Comment by Clockwork-Muse** [comment]
@stephentoub Addressed and pushed.

**Comment by stephentoub** [comment]
Thanks

**Pull request merged**